### PR TITLE
Add delegated_type to ActiveRecord::Base

### PR DESCRIPTION
**Pull request by ksss**
`delegated_type` is a one of public API in ActiveRecord.

https://edgeapi.rubyonrails.org/classes/ActiveRecord/DelegatedType.html

### CODE REVIEW

**Comment by ksss** [comment]
OMG, `delegated_type` was an API from 6.1.
How do you think in this case?

**Comment by soutaro** [comment]
Copy-on-write! 🐮

**Comment by ksss** [comment]
Thank you for reviewing.
I'll try 🐮.

**Comment by ksss** [comment]
I created `6.1` dir and set symlink from `6.0`.
Please check it out!

File: gems/activerecord/6.1/_test/Steepfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  library "activerecord:6.1.4.1"
+  library "activesupport:6.1.4.1"
+  library "actionpack:6.1.4.1"
+  library "activejob:6.1.4.1"
+  library "activemodel:6.1.4.1"
+  library "actionview:6.1.4.1"
+  library "railties:6.1.4.1"
+
+  typing_options :strict
```
Comment (pocke) [inline gems/activerecord/6.1/_test/Steepfile:24]: Could you use the new configuration of Steep which is introduced v0.46.0? For example: https://github.com/ruby/gem_rbs_collection/pull/58


The other parts of this PR looks good :+1:

**Comment by ksss** [comment]
Rebase and updated!

**Review by pocke — APPROVED** [review]
LGTM :star2: Thanks!

**Pull request merged**